cmd/explaintest/r/naaj.result

```diff
@@ -0,0 +1,410 @@
+delete from naaj_A;
+delete from naaj_B;
+insert into naaj_A values(1,null,1);
+select (a, b) not in (select a, b from naaj_B) from naaj_A;
+(a, b) not in (select a, b from naaj_B)
+1
+select * from naaj_A where (a, b) not in (select a, b from naaj_B);
+a	b	c
+1	NULL	1
+insert into naaj_B values(2, null, 2);
+select (a, b) not in (select a, b from naaj_B) from naaj_A;
+(a, b) not in (select a, b from naaj_B)
+1
+select * from naaj_A where (a, b) not in (select a, b from naaj_B);
+a	b	c
+1	NULL	1
+insert into naaj_B values(null, null, 2);
+select (a, b) not in (select a, b from naaj_B) from naaj_A;
+(a, b) not in (select a, b from naaj_B)
+NULL
+select * from naaj_A where (a, b) not in (select a, b from naaj_B);
+a	b	c
+delete from naaj_B;
+insert into naaj_B values(2, 2, 2);
+select (a, b) not in (select a, b from naaj_B) from naaj_A;
+(a, b) not in (select a, b from naaj_B)
+1
+select * from naaj_A where (a, b) not in (select a, b from naaj_B);
+a	b	c
+1	NULL	1
+insert into naaj_B values(2, null, 2);
+insert into naaj_B values(null, null, 2);
+explain format = 'brief' select (a, b) not in (select a, b from naaj_B where naaj_A.c > naaj_B.c) from naaj_A;
+id	estRows	task	access object	operator info
+HashJoin	10000.00	root		Null-aware anti left outer semi join, equal:[eq(test.naaj_a.b, test.naaj_b.b) eq(test.naaj_a.a, test.naaj_b.a)], other cond:gt(test.naaj_a.c, test.naaj_b.c)
+├─TableReader(Build)	10000.00	root		data:TableFullScan
+│ └─TableFullScan	10000.00	cop[tikv]	table:naaj_B	keep order:false, stats:pseudo
+└─TableReader(Probe)	10000.00	root		data:TableFullScan
+  └─TableFullScan	10000.00	cop[tikv]	table:naaj_A	keep order:false, stats:pseudo
+select (a, b) not in (select a, b from naaj_B where naaj_A.c > naaj_B.c) from naaj_A;
+(a, b) not in (select a, b from naaj_B where naaj_A.c > naaj_B.c)
+1
+explain format = 'brief' select * from naaj_A where (a, b) not in (select a, b from naaj_B  where naaj_A.c > naaj_B.c);
+id	estRows	task	access object	operator info
+HashJoin	8000.00	root		Null-aware anti semi join, equal:[eq(test.naaj_a.b, test.naaj_b.b) eq(test.naaj_a.a, test.naaj_b.a)], other cond:gt(test.naaj_a.c, test.naaj_b.c)
+├─TableReader(Build)	10000.00	root		data:TableFullScan
+│ └─TableFullScan	10000.00	cop[tikv]	table:naaj_B	keep order:false, stats:pseudo
+└─TableReader(Probe)	10000.00	root		data:TableFullScan
+  └─TableFullScan	10000.00	cop[tikv]	table:naaj_A	keep order:false, stats:pseudo
+select * from naaj_A where (a, b) not in (select a, b from naaj_B  where naaj_A.c > naaj_B.c);
+a	b	c
+1	NULL	1
+explain format = 'brief' select (a, b) not in (select a, b from naaj_B where naaj_A.c = naaj_B.c) from naaj_A;
+id	estRows	task	access object	operator info
+HashJoin	10000.00	root		anti left outer semi join, equal:[eq(test.naaj_a.c, test.naaj_b.c)], other cond:eq(test.naaj_a.a, test.naaj_b.a), eq(test.naaj_a.b, test.naaj_b.b)
```

Comment:
yes, it does.

here is the old path, since it already has the correlated EQ condition "naaj_A.c = naaj_B.c", which could be used as a hash join key. So the na-eq condition should be put in the other condition as it was before. (mixture EQ cond and NA-EQ cond in hash join key is little tricky)

Reply:
This can be future improvements. We can open another issue to track this optimization (EQ and NAEQ as the hash keys).

Reply:
make sense